dj/sql/functions.py

```diff
@@ -1277,6 +1277,20 @@ def infer_type(  # noqa: F811
     return ct.ListType(element_type=element_type)
 
 
+class Array_Append(Function):  # pylint: disable=abstract-method,disable=invalid-name
+    """
+    Add the element at the end of the array passed as first argument
+    """
+
+
+@Array_Append.register  # type: ignore
+def infer_type(  # noqa: F811
+    array: ct.ListType,
+    item: ct.ColumnType,
+) -> ct.ListType:
+    return ct.ListType(element_type=item.type)
```

Comment:
Is it up to the engine to raise if the array and item have different types

Reply:
That's what I was thinking. I started to validate that here but realized it would be duplicating what the engine will do anyway. If this is used with an engine that supports multi-type arrays, that would raise a different issue with the type inference but I can't think of an engine that does allow that. Maybe that's just be a bridge we cross when we get to it.